GithubProfileWiki/Networking/APIPath.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  APIPath.swift
+//  GithubProfileWiki
+//
+//  Created by ilter on 23.01.2022.
+//
+
+import Foundation
+
+#if DEBUG
+let environment = Environment.development
+#else
+let environment = Environment.production
+#endif
+
+
+private let baseURL = environment.baseUrl()
+
+struct Path {
+    
+    var userName: String
+    
+    var path: String {
+            return "\(baseURL)/users/\(userName)/followers"
```

Comment:
i think you should not add baseUrl to path of each request. if it is really your base url, you should handle that in the network layer

Reply:
Actually, this is the fully generic and dynamic network layer, Maybe I won't just use GitHub API, I might add GitLab or Bitbucket that's why I made it like this after your comment? But Still thinking because I initialize a struct every call.

What do you think about this?

```
struct Environment {
    
    var method: String {
        "https://"
    }
    
    var domain: String {
        "github.com"
    }
    
    var subdomain: String {
        "api"
    }
    
    var baseUrl: String {
        "\(method)\(subdomain).\(domain)"
    }
}
```

```
struct URLPath {
    
    var userName: String
    
    var followersUrl: String {
        return "\(Environment().baseUrl)/users/\(userName)/followers"
    }
}
```